src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,478 @@
+
+    if input_cores and input_accessions is None:
+        query_accessions = fetch_accessions_from_cores(query_list, partial_url)
+        query_type = "CoreDB"
+    elif input_cores is None and input_accessions:
+        query_type = "Accession"
+        for accession in query_list:
+            match = re.match(r"(GC[AF])_([0-9]{3})([0-9]{3})([0-9]{3})\.?([0-9]+)", accession)
+            if not match:
+                raise UnsupportedFormatError(f"Could not recognize GCA accession format: {accession}")
+            query_accessions[accession] = accession
+
+    return query_accessions, query_type
+
+
+def fetch_accessions_from_cores(database_names: list, connection_url: URL) -> Dict:
+    """Obtain the associated INSDC accession [meta.assembly.accession] given a set of core(s) names
+    and a MYSQL server host.
+
+    Args:
+        database_names: Set of names for one or more core databases
+        connection_url: Partial MYSQL host name : port
+    Returns:
+        Dict of core name(s) (key) and its INSDC assembly.accession (value)
+    """
+
+    core_accn_meta = {}
+    core_list_count = len(database_names)
+    count_accn_found = 0
+
+    for core in database_names:
+        db_connection_url = connection_url.set(database=core)
+        db_connection = dbc(f"{db_connection_url}")
+        qry_result = db_connection.execute(
+            'SELECT meta_value FROM meta WHERE meta_key = "assembly.accession";'
+        ).fetchall()
+
+        if qry_result is None:
+            logging.warning(f"We have no accession on core: {core}")
+        elif len(qry_result) == 1:
+            count_accn_found += 1
+            asm_accession = qry_result.pop()[0]
+            logging.info(f"{core} -> assembly.accession[{asm_accession}]")
+            core_accn_meta[core] = asm_accession
+        else:
+            logging.warning(f"Core {core} Has {len(qry_result)} assembly.accessions")
+
+    logging.info(f"From initial input cores ({core_list_count}), obtained ({count_accn_found}) accessions")
+
+    return core_accn_meta
+
+
+def datasets_asm_reports(
+    sif_image: str, assembly_accessions: dict, download_directory: PathLike, batch_size: int
+) -> Dict:
+    """Obtain multiple assembly report JSONs in one or more querys to datasets,
+    i.e. make individual since accn query to datasets tool.
+
+    Args:
+        sif_image: Instance of Client.loaded singularity image.
+        assembly_accessions: Dict of core accessions.
+        download_directory: Dir path to store assembly report JSON files.
+        batch_size: Number of assembly accessions to batch submit to 'datasets'.
+
+    Returns:
+        Dictionary of core name and its associated assembly report
+    """
+
+    master_accn_list = list(assembly_accessions.values())
+    combined_asm_reports = {}
+
+    # Setting the number of combined accessions to query in a single call to datasets
+    list_split = list(range(0, len(master_accn_list), batch_size))  ## Note best to use >=10
```

Comment:
```suggestion
    list_split = list(range(0, len(master_accn_list), batch_size))
```
If there is an optimal/suggested number, I would set it as the default value of `batch_size`. Furthermore, I would add this comment to the docstring rather than here, so this information is available to users.